flink-scala/src/main/scala/org/apache/flink/api/scala/metrics/ScalaGauge.scala

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.api.scala.metrics
+
+import org.apache.flink.metrics.Gauge
+
+class ScalaGauge[T](value : T) extends Gauge[T] {
```

Comment:
Directly assigning value in constructor is one possible approach to achieve the goal of reducing the minimal code lines needed for registering a Scala gauge. However, when we meet situations like it will contain some logics to calculate the value, this approach will not work.
One possible solution is to use Scala's Function0 utility to pass in a "value generator" function instead of value itself. Something like this:
`
class ScalaGauge[T](func: () => T) extends Gauge[T]
`
then one can use the class like this:
`
val myGauge = new ScalaGauge(() => 1)
`
or
`
    val myGauge = new ScalaGauge(() => {
      // codes to calculate the value
    })
`
when we meet some complex situations.

Here is the Function0 reference which described the usage more clearly:
http://www.scala-lang.org/api/2.6.0/scala/Function0.html